src/js/load-package.ts

```diff
@@ -297,7 +312,7 @@ async function loadDynlib(lib: string, shared: boolean) {
       });
     }
   } catch (e: any) {
-    if (e.message.includes("need to see wasm magic number")) {
+    if (e && e.message && e.message.includes("need to see wasm magic number")) {
       console.warn(
         `Failed to load dynlib ${lib}. We probably just tried to load a linux .so file or something.`
```

Comment:
Not related to this PR: another possibility that we might want to mention in this error message is that `-sSIDE_MODULE` went missing and the file `blah.so` actually contains JavaScript.

Reply:
Yes, indeed. It is a pretty common reason of the failure.